peps/pep-0725.rst

```diff
@@ -377,8 +339,67 @@ consistency between Python dependencies and external dependencies, we choose to
+  Python specifier (distribution), but a ``dep:`` URL.
+- ``Provides-External-Extra``. An *extra* group that carries external dependencies
+  (as found in ``Requires-External-Dep``) only.
+
+Since the Core Metadata specification does not contain fields for any metadata in
+``pyproject.toml``'s ``[build-system]`` table, the ``build-requires``
+and ``host-requires`` content do not need to be reflected in existing core
+metadata fields.
+
+Additionally, this PEP also proposes to deprecate the ``Requires-External`` field.
+The reasons being:
+
+- Avoiding confusion with the newly proposed fields.
+- Avoiding potential incompatibilities with existing usage (even if limited).
+- Low penetration in the ecosystem:
+
+  - There is no direct correspondence to a field in the ``pyproject.toml`` metadata.
+  - Mainstream build backends like ``setuptools`` (see `pypa/setuptools#4220`_),
+    ``hatch`` (see `pypa/hatch#1712`_), ``flit`` (see `pypa/flit#353`_), or ``poetry``
+    do not offer ways to specify it or require a plugin (e.g. `poetry-external-dependencies`_).
+    ``maturin`` does seem to support it since 0.7.0 (see `PyO3/maturin@5b0e4808`_),
+    but it's not directly documented. Other backends like ``scikit-build-core`` or
+    ``meson-python`` returned no results for ``External-Requires``.
+  - The field is not included in the `PyPI JSON API responses`_.
+
+Note about the differences between sdist and wheel metadata
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+A wheel may vendor its external dependencies. This happens in particular when
+distributing wheels on PyPI or other Python package indexes -- and tools like
+auditwheel_, delvewheel_ and delocate_ automate this process. As a result, a
+``Requires-External`` entry in an sdist may disappear from a wheel built from
+that sdist. It is also possible that a ``Requires-External`` entry remains in a
+wheel, either unchanged or with narrower constraints. ``auditwheel`` does not
+vendor certain allow-listed dependencies, such as OpenGL, by default. In
+addition, ``auditwheel`` and ``delvewheel`` allow a user to manually exclude
+dependencies via a ``--exclude`` or ``--no-dll`` command-line flag. This is
+used to avoid vendoring large shared libraries, for example those from CUDA.
+
+``Requires-External`` entries generated from external dependencies in
+``pyproject.toml`` MAY differ between an sdist and its corresponding wheel(s).
+However, the ``Requires-External`` entries in the wheels MUST always specify
+a narrower set of dependencies.
+
+Note that this does not imply that the field must be marked as Dynamic, since this
+distinction only applies to wheels built from an sdist by a build backend. In
+particular, wheels built from other wheels do not need to satisfy this constraint (see
+`message 179 in this DPO thread
+<https://discuss.python.org/t/split-from-pep-639-expressing-project-vs-distribution-licenses-post-pep-639-mod-titled/90314/179>`__
```

Comment:
I'll note that this isn't reflected in the actual spec yet, so including it here means we'll probably have to circle back to that. But it's fine for that to happen later.